src/Microsoft.VisualStudio.Services.Agent/ProcessInvoker.cs

```diff
@@ -545,31 +556,65 @@ public uint Size
         // Delegate type to be used as the Handler Routine for SetConsoleCtrlHandler
         private delegate Boolean ConsoleCtrlDelegate(ConsoleCtrlEvent CtrlType);
 #else
-        private async Task NixCancelAndKillProcessTree()
+        private async Task<bool> SendSignal(Signals signal, TimeSpan timeout)
         {
-            // TODO: replace Task.Delay(1) with Send SIGINT/SIGTERM/SIGKILL
-            await Task.Delay(1);
-            KillProcessTree();
+            Trace.Info($"Sending {signal} to process {_proc.Id}.");
+            try
+            {
+                int errorCode = kill(_proc.Id, (int)signal);
+                if (errorCode != 0)
+                {
+                    throw new Win32Exception(errorCode);
```

Comment:
Umm? It looks like this exception will get eaten by the catch handler below, but still this is a strange type to use from code that only runs on OSX/Linux.